src/site/stages/build/plugins/inject-vale-linter.js

```diff
@@ -0,0 +1,160 @@
+/* eslint-disable no-continue, no-unused-vars */
+
+/**
+ * TODO: Handle errors with sentry where appropriate, or find out what's best todo with these errors
+ */
+
+/**
+ * Run vale lint on the page content and inject suggestions into the rendered HTML
+ */
+const path = require('path');
+const { spawn } = require('child_process');
+const fs = require('fs');
+const tmp = require('tmp');
+
+/**
+ *  Method to execute vale check on a content string via helper script
+ *
+ * @param {String} content
+ */
+async function runValeCheck(content) {
+  const promise = new Promise((resolve, _reject) => {
+    const results = {};
+
+    /**
+     * TODO: We need to ensure some sort of check so bad content can't be passed in
+     */
+    const vale = spawn('vale', [
+      '--config=script/vale/.vale.ini',
+      '--output=JSON',
+      `${content}`,
+    ]);
+
+    vale.stdout.on('data', data => {
+      results.results = data;
+      results.dataString = data.toString();
+    });
+
+    vale.stderr.on('data', data => {
+      results.errors = data.toString();
+      // TODO log error somewhere
+      // console.log(`vale stderr: ${data.toString()}`);
+    });
+
+    vale.on('exit', code => {
+      results.code = code.toString();
+      // TODO handle errors
+      // console.log(`child process exited with code ${code.toString()}`);
+    });
+
+    resolve(results);
+  });
+  const results = await promise;
+  await new Promise((resolve, _reject) => {
+    setTimeout(resolve, 5000);
+  }); // Only works with this
+  return results;
+}
+
+/**
+ * Create the vale temp file
+ * @param {Buffer} dataBuffer the content to write to file
+ */
+function createTempFile(dataBuffer) {
+  const temp = tmp.fileSync({ prefix: 'vale-' });
+  fs.writeSync(temp.fd, dataBuffer);
+  return temp.name;
+}
+
+/**
+ * Inject the results into the page header banner element
+ */
+function injectResultsIntoMarkup(file, issues) {
+  const { dom } = file;
+  const bannerEl = document.createElement('div');
```

Comment:
So I think you'll need to use [Cheerio](https://cheerio.js.org) here, since `dom` comes from parsing the HTML string using that library. This might be why you aren't seeing it show up in your tests.